bib/internal/utils/fileutils.go

```diff
@@ -0,0 +1,14 @@
+package utils
+
+import (
+	"os"
+
+	"github.com/sirupsen/logrus"
+)
+
+// LogClose closes the file and logs any error encountered during the operation.
+func LogClose(file *os.File) {
+	if err := file.Close(); err != nil {
+		logrus.WithError(err).Errorf("failed to close file")
+	}
+}
```

Comment:
Do we need this?  An inline deferred function wouldn't be too bad.  From this diff it seems we only need it in a handful of places.  And having it inline would also let us add more context to each error (like `failed to close config file`).